### PR TITLE
Tiny fixes for generator stability.

### PR DESCRIPTION
**Pull request by jmacd**
## What is the current behavior?

Crashes from a nil pointer.

## What is the new behavior?
In certain scenarios, a zero is returned where it would crash before.

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests(`make test`) for the changes have been added (for bug fixes / features) and pass
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Lint (`make lint`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: generatorreceiver/internal/topology/latency_percentiles.go [hunk 1]
```diff
@@ -29,20 +29,26 @@ type LatencyPercentiles struct {
...
+	}
 	uniform := func(timeA, timeB time.Duration) int64 {
 		min := float64(timeA.Nanoseconds())
 		max := float64(timeB.Nanoseconds())
 		return int64(min + (max-min)*rand.Float64())
 	}
 	genNumber := rand.Float64()
 	switch {
-	case genNumber <= 0.5:
+	case genNumber < 0.5:
```
Comment (jmacd) [inline generatorreceiver/internal/topology/latency_percentiles.go:45]: Editorial nits: the Float64() range is [0, 1) so the comparison should be `<` and not `<=` not that it really matters. 😂

File: generatorreceiver/internal/topology/pickable.go [hunk 2]
```diff
@@ -29,23 +30,30 @@ func pickBasedOnWeight[P Pickable](ps []P, traceID pcommon.TraceID) P {
...
 	traceUint := binary.BigEndian.Uint64(secondHalf)
-	// Use the last two digits as percentage.
-	choice := float64(traceUint%100) / 100.0
-
-	choice = choice * totalWeight
+	// Use the half of the traceID as a ratio.
+	ratio := float64(traceUint) / float64(math.MaxUint64)
+	// Search for the item by weight from N-1 items.
+	chooseFrom := activeSets[:len(activeSets)-1]
+	choice := ratio * totalWeight
```
Comment (jmacd) [inline generatorreceiver/internal/topology/pickable.go:47]: I.d.k. this looks nicer to me.

**Review by kristinapathak — APPROVED** [review]

**Review by paivagustavo — APPROVED** [review]
I think the crashes has been fixed by https://github.com/lightstep/telemetry-generator/pull/45
but I like extra cautions :)

**Comment by jmacd** [comment]
Surprised no merge conflicts 😂

**Pull request merged**